### DISCUSSION TITLE
Idea: plugins and/or access to form lifecycle methods

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#7430] [Ideas]
Hi all I wanted to open a discussion around the idea of adding some sort of plugins or access to lifecycle methods to react-hook-form. The idea being as the library grows there are more and more feature requests being made. Many of these feature requests might not align with the core library goals or be difficult to implement. One solution to this problem might be to add some sort of plugin system or read-only access to the form internal state via lifecycle callbacks.

With this system users could extend the react hook form functionality without adding additional code to the core library. This would also take the load off adding new features as users could safely share plugins and whatnot without jeopardizing the core library functionality. 

This might be overkill for this library but I just wanted to share this idea and see what you all thought. 

@jorisre @bluebill1049 @barrymay

**Comment by bluebill1049** [comment]
thanks @wdfinch that's why the expose utility props was one of the option, do you have any idea or example for such API?

**Reply by wdfinch** [reply]
Not yet, I just wanted to see what you thought about the idea.
